lib/active_merchant/billing/gateways/globalcollect.rb

```diff
@@ -0,0 +1,294 @@
+            "additionalInfo" => billing_address[:address2],
+            "zip" => billing_address[:zip],
+            "city" => billing_address[:city],
+            "state" => billing_address[:state],
+            "countryCode" => billing_address[:country]
+          }
+        end
+        if shipping_address
+          post["order"]["customer"]["shippingAddress"] = {
+            "street" => shipping_address[:address1],
+            "additionalInfo" => shipping_address[:address2],
+            "zip" => shipping_address[:zip],
+            "city" => shipping_address[:city],
+            "state" => shipping_address[:state],
+            "countryCode" => shipping_address[:country]
+          }
+          post["order"]["customer"]["shippingAddress"]["name"] = {
+            "firstName" => shipping_address[:firstname],
+            "surname" => shipping_address[:lastname]
+          }
+        end
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def url(action, authorization)
+        (test? ? test_url : live_url) + uri(action, authorization)
+      end
+
+      def uri(action, authorization)
+        uri = "/v1/#{@options[:merchant_id]}/"
+        case action
+        when :test_connection
+          uri + "services/testconnection"
+        when :authorize
+          uri + "payments"
+        when :capture
+          uri + "payments/#{authorization}/approve"
+        when :refund
+          uri + "payments/#{authorization}/refund"
+        when :void
+          uri + "payments/#{authorization}/cancel"
+        end
+      end
+
+      def commit(action, post, authorization = nil)
+        begin
+        response = parse(ssl_post(url(action, authorization), post.to_json, headers(action, post, authorization)))
+        rescue ResponseError => e
+          raise unless(e.response.code.to_s =~ /4\d\d/)
+          response = parse(e.response.body)
+        end
+
+        succeeded = success_from(response)
+        Response.new(
+        succeeded,
+        message_from(succeeded, response),
+        response,
+        authorization: authorization_from(succeeded, response),
+        error_code: error_code_from(succeeded, response),
+        test: test?
+        )
+
+      end
+
+      def headers(action, post, authorization = nil)
+        {
+          "Content-type"  => content_type,
+          "Authorization" => auth_digest(action, post, authorization),
+          "Date" => date
+        }
+      end
+
+      def auth_digest(action, post, authorization = nil)
+        data = <<-EOS
+POST
+#{content_type}
+#{date}
+#{uri(action, authorization)}
+EOS
+        digest = OpenSSL::Digest.new('sha256')
+        key = @options[:secretApiKey]
+        "GCS v1HMAC:#{@options[:apiKeyId]}:#{Base64.strict_encode64(OpenSSL::HMAC.digest(digest, key, data))}"
+      end
+
+      def date
+        @date ||= Time.now.strftime("%a, %d %b %Y %H:%M:%S") + " GMT" # Must be same in digest and HTTP header
```

Comment:
Instead of hard-coding the timezone, we should probably do:

```
Time.now.utc.strftime("%a, %d %b %Y %H:%M:%S %Z")
```

Otherwise, the timestamp will actually be off by 4-5 hours (the diff between GMT and EDT - the local timezone)